kdump/include/runtest.sh

```diff
@@ -511,7 +506,82 @@ AppendConfig()
         shift
     done
 
-    rstrnt-report-log -l "${KDUMP_CONFIG}"
+    RhtsSubmit "${KDUMP_CONFIG}"
+}
+
+
+AppendSysconfig()
+{
+    Log "- Modifying /etc/sysconfig/kdump"
+
+    local KEY=$1
+    local ACTION=$2
+    local VALUE1=$3
+    local VALUE2=$4
+
+    if [ -z "$KEY" ] || [ -z "$ACTION" ] || [ -z "$VALUE1" ]; then
+        Error "- Missing KEY or ACTION or VALUE1."
+        return 1
+    elif [ "$ACTION" = "replace" ] && [ -z "$VALUE2" ]; then
+        Error "- Missing new_value for replacing."
+        return 1
+    elif ! grep "^$KEY=\"" "${KDUMP_SYS_CONFIG}"; then
+        Error "- Invalid KEY: $KEY."
+        return 1
+    fi
+
+    local kdump_sys_config_tmp="${KDUMP_SYS_CONFIG}.tmp"
+    \cp "${KDUMP_SYS_CONFIG}" "${kdump_sys_config_tmp}"
+
+    # Note: When KEY is "KDUMP_COMMANDLINE", ACITION add/remove/replace is actually
+    # manipulating the value of current kernel cmdline and assign
+    # it to KDUMP_COMMANDLINE.
+    # if there is no value set to KDUMP_COMMANLINE, assign current kernel cmdline
+    # to it for later string manipluation.
+    if [ "$KEY" == "KDUMP_COMMANDLINE" ] && \
+        grep "^$KEY=\"[[:space:]]*\"" "${kdump_sys_config_tmp}"; then
+
+        sed -i "/^KDUMP_COMMANDLINE=\"/d" "${kdump_sys_config_tmp}"
+        echo "KDUMP_COMMANDLINE=\"$(cat /proc/cmdline)\"" >> "${kdump_sys_config_tmp}"
+    fi
+
+    case $ACTION in
+        add)
+            # Check if the key already has a value on it, then append " $VALUE1" to the
+            # original value.
+            if grep "^$KEY=\"[[:space:]]*\"" "${kdump_sys_config_tmp}"; then
+                sed -i /^"$KEY="/d "${kdump_sys_config_tmp}"
+                echo "$KEY=\"$VALUE1\"" >> "${kdump_sys_config_tmp}"
+            else
+                sed -i  /^"$KEY="/s/\"/" $VALUE1"\"/2 "${kdump_sys_config_tmp}"
```

Comment:
I'm not sure I understand this line, but wouldn't it be more readable like this?
```
sed -i  "/^$KEY=/s/\"/ $VALUE1\"/2" "${kdump_sys_config_tmp}"
```
Also, wouldn't that replace the **first** double quote, instead of the last? Shouldn't the regular expression be `\"$`, with the dollar at the end?

Reply:
The original idea is to replace the second (which usually the last quote) as kernel options are normally provided without extra quotes. But you are right, \"$ would be more accurate. Fixed.